schedule/alg_data_generator.py

```diff
@@ -85,7 +85,13 @@ def merge_preferred_times(unmerged_preferred_times):
         "spring": {},
         "summer": {}
     }
+
     for semester in unmerged_preferred_times:
+        #handle edge case where some semester may not have any preferredTimes - must set to None
+        if unmerged_preferred_times[semester] == 'null':
```

Comment:
I think this should be set to None, as they send us null which I think Django converts to None when I tested it.

Reply:
You could always do `if not unmerged_preferred_times[semester]` too

Reply:
Hm let me try this one once more. I'm pretty certain Django holds this as a `null` field. Will check it out now

Reply:
Fixed to check for any falsy value